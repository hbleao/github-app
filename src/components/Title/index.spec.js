/* eslint-disable no-use-before-define */
import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Title } from '.';

const makeSut = ({ children, className }) => {
  render(<Title className={className}>{children}</Title>);
};

describe('Title', () => {
  it('should be able render correctly', () => {
    makeSut({ children: 'Title default' });

    const titleTextDefault = screen.getByText(/Title default/gi);
    expect(titleTextDefault).toBeTruthy();
  });

  it('should be able to render component with text confirm', () => {
    makeSut({ children: 'Github Actions' });

    const titleTextDefault = screen.getByText(/github actions/gi);
    expect(titleTextDefault).toBeTruthy();
  });

  it('should be able to render component with custom className', () => {
    const className = 'custom-class';
    const children = 'Github Actions';

    makeSut({ className, children });

    const titleWithCustomClassName = document.querySelector(`.${className}`);
    expect(titleWithCustomClassName).toBeTruthy();
  });

  it('should be able to render snapshot', () => {
    const component = renderer.create(<Title>Github app</Title>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
