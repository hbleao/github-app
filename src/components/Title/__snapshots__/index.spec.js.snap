// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should be able to render snapshot 1`] = `
<h1
  className="title "
>
  Github app
</h1>
`;
